File: modules/exploratory_data_analysis/data_management_in_R/script_04.R

library(dplyr)

salary_data <- read.csv("../../../data/data_management/sal_data.csv", header = TRUE)
head(salary_data)

bonus_data <- read.csv("../../../data/data_management/bonus_data.csv", header = TRUE)
head(bonus_data)


inner_join(salary_data, bonus_data, by = "Employee_ID")
left_join(salary_data, bonus_data, by = "Employee_ID")
right_join(salary_data, bonus_data, by = "Employee_ID")
full_join(salary_data, bonus_data, by = "Employee_ID")
semi_join(salary_data, bonus_data, by = "Employee_ID")
anti_join(salary_data, bonus_data, by = "Employee_ID")

File: modules/0104_exploratory_data_analysis/00_live_class/script_04.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Sat Oct  7 10:30:26 2023

@author: jerry
"""


import pandas as pd


salary_data = pd.read_csv("../../../data/eda/data_management/sal_data.csv")
bonus_data = pd.read_csv("../../../data/eda/data_management/bonus_data.csv")


left_join = pd.merge(salary_data, bonus_data, how = 'left')
left_join


right_join = pd.merge(salary_data, bonus_data, how = 'right')
right_join 


inner_join = pd.merge(salary_data, bonus_data)
inner_join 


outer_join = pd.merge(salary_data, bonus_data, how = 'outer')
outer_join 
